app/models/task.py

```diff
@@ -1,5 +1,36 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
+from datetime import datetime
+from typing import Optional
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING: from .goal import Goal
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[Optional[datetime]] = mapped_column(nullable=True)
+    goal_id: Mapped[Optional[int]] = mapped_column(db.ForeignKey("goal.id"))  
+    goal: Mapped[Optional["Goal"]] = relationship("Goal", back_populates="tasks")  
+
+    @classmethod
+    def from_dict(cls, task_data):
+        new_task = Task(
+            title=task_data["title"],
+            description=task_data["description"],
+            completed_at=task_data.get("completed_at")
+            
+        )
+        return new_task
```

Comment:
I left the same comment for the `from_dict` class method in `goal.py`. 

I'd prefer you use the `cls` keyword here instead of `Task` by name to keep this method more generic. If you ever had to change this class name to something else, then you would need to remember to refactor line 19 too. 

```suggestion
        return cls(
            title=task_data["title"],
            description=task_data["description"],
            completed_at=task_data.get("completed_at")
        )
```